apps/settings/test/unit/panels/screen_lock_passcode/screen_lock_passcode_test.js

```diff
@@ -111,10 +114,16 @@ suite('ScreenLockPasscode > ', function() {
     });
 
     suite('passcode is different', function() {
-      setup(function() {
-        screenLockPasscode._settings.passcode = '0123';
-        screenLockPasscode._passcodeBuffer = '0000';
-        screenLockPasscode._checkPasscode();
+      setup(function(done) {
+        // set passcode to 0123
+        var promise = PasscodeHelper.set('0123');
```

Comment:
Because `PasscodeHelper` is a mock, we can make it return any value that we need for the test case. This can be done by using `this.sinon.stub`. Here is how:
```
this.sinon.stub(PasscodeHelper, 'check').returns(Promise.resolve(false));
```
Doing this saves lots of work of setting things up and you can then write assertions directly.